app/models/mark.rb

```diff
@@ -18,6 +19,44 @@ class Mark < ApplicationRecord
   validates_uniqueness_of :markable_id,
                           scope: [:result_id, :markable_type]
 
+  validates_inclusion_of :override, in: [true, false]
+
+  def calculate_deduction
+    return 0 if self.markable_type != 'FlexibleCriterion' || self.override?
```

Comment:
Possibly remove `|| self.override` as long as every place calling this method checks that.

Reply:
This helper method should stand on its own for the relevant conditions and not be dependent on existing calls or calls implemented in the future to check this condition.